src/wms/src/main/java/org/geoserver/wms/icons/IconService.java

```diff
@@ -49,7 +49,7 @@ public ModelAndView handleRequestInternal(
                     400, "Bad request, path must be of form: /icons/[<workspace>/]<style>");
             return null;
         }
-
+//this is wrong - matches 2 even when no workspace in url!
```

Comment:
JIRA task?

Reply:
I'm scared to fix it with out checking the existing KML test coverage, it does work but is misleading.